src/joint.dia.paper.js

```diff
@@ -1374,27 +1374,84 @@ joint.dia.Paper = joint.mvc.View.extend({
         return this;
     },
 
-    drawGrid: function(opt) {
+    getGridData: function (opt) {
+
+        var svgGridElements = this._svgGridData;
 
-        opt = _.defaults({}, opt, this.options.drawGrid);
+        if (!svgGridElements) {
+
+            svgGridElements = {
+                xmlSerializer: new XMLSerializer(),
+                pattern: V('<pattern id="grid-pattern" patternUnits="userSpaceOnUse"></pattern>'),
+                gridShape: opt.gridPattern ? V(opt.gridPattern) : V('<rect />').attr('fill', opt.color)
```

Comment:
Can we move "dot" logic to default options as  `gridPattern` and `updateGridPattern`?